python/lsst/ts/salkafka/kafka_producer_factory.py

```diff
@@ -89,7 +100,20 @@ def __init__(self, config, log):
         self.http_session = None  # created by `start`
         self.schema_registry = None
         self.log.info("Making Kafka client session")
-        self.broker_client = AdminClient({"bootstrap.servers": self.config.broker_url})
+        if self.config.sasl_plain_username:
```

Comment:
What does AdminClient do if username or password are None? Unless it presents a really clear error message, please check here.